src/chk/chk_leader.c

```diff
@@ -3346,10 +3398,20 @@ chk_leader_query(int pool_nr, uuid_t pools[], chk_query_head_cb_t head_cb,
 		d_iov_set(&riov, NULL, 0);
 		d_iov_set(&kiov, cpr->cpr_uuid, sizeof(uuid_t));
 		rc = dbtree_lookup(ins->ci_pool_hdl, &kiov, &riov);
-		if (likely(rc == 0))
-			tmp = (struct chk_pool_rec *)riov.iov_buf;
-		else
-			tmp = NULL;
+		if (likely(rc == 0)) {
+			status = ((struct chk_pool_rec *)riov.iov_buf)->cpr_bk.cb_pool_status;
+			phase  = ((struct chk_pool_rec *)riov.iov_buf)->cpr_bk.cb_phase;
+		} else {
+			uuid_unparse_upper(cpr->cpr_uuid, uuid_str);
```

Comment:
suggestion - it may be clearer to wrap calls to these uuid_unparse_upper/lower functions with descriptively named functions like `chk_uuid_unparse_leader` and `chk_uuid_unparse_follower`.